File: lib/features/saliks/domain/entities/area.dart
import '../../../../core/utils/text_field_merge.dart';
import 'bazam.dart';

class Area {
  const Area({
    required this.areaId,
    required this.areaName,
    this.bazamId = kDefaultBazamId,
  });

  final String areaId;
  final String areaName;
  final String bazamId;

  factory Area.fromMap(Map<String, dynamic> map, {String? id}) {
    final areaId = (map['areaId'] as String?) ?? id;
    if (areaId == null || areaId.isEmpty) {
      throw const FormatException('Area document missing areaId');
    }
    final rawBazam = (map['bazamId'] as String?)?.trim() ?? '';
    return Area(
      areaId: areaId,
      areaName: mergeLegacyBilingual(
        primary: map['areaName'] as String? ?? '',
        secondary: map['areaNameUrdu'] as String? ?? '',
      ),
      bazamId: rawBazam.isEmpty ? kDefaultBazamId : rawBazam,
    );
  }

  Map<String, dynamic> toMap() {
    final split = splitBilingualLabel(areaName);
    return {
      'areaId': areaId,
      if (split.english.isNotEmpty) 'areaName': split.english,
      if (split.urdu.isNotEmpty) 'areaNameUrdu': split.urdu,
      'bazamId': bazamId.isEmpty ? kDefaultBazamId : bazamId,
    };
  }
}

File: app/src/main/java/com/example/salik_management_system/features/saliks/domain/model/Area.kt
package com.example.salik_management_system.features.saliks.domain.model

import com.example.salik_management_system.features.saliks.domain.model.kDefaultBazamId

data class Area(
    val areaId: String,
    val areaName: String,
    val bazamId: String = kDefaultBazamId,
) {
    fun toMap(): Map<String, Any?> = mapOf(
        "areaId" to areaId,
        "areaName" to areaName,
        "bazamId" to bazamId.ifEmpty { kDefaultBazamId },
    )

    companion object {
        fun fromMap(map: Map<String, Any?>, id: String? = null): Area {
            val areaId = (map["areaId"] as? String) ?: id
            require(!areaId.isNullOrEmpty()) { "Area document missing areaId" }
            val rawBazam = (map["bazamId"] as? String)?.trim().orEmpty()
            val primary = (map["areaName"] as? String).orEmpty()
            val secondary = (map["areaNameUrdu"] as? String).orEmpty()
            val name = when {
                primary.isNotEmpty() && secondary.isNotEmpty() && primary != secondary ->
                    "$primary / $secondary"
                primary.isNotEmpty() -> primary
                else -> secondary
            }
            return Area(
                areaId = areaId,
                areaName = name,
                bazamId = rawBazam.ifEmpty { kDefaultBazamId },
            )
        }
    }
}
